### ISSUE TITLE
Change fingerprint hash from sha1 to sha256

### ISSUE THREAD

**Issue by nyxnor** [raspiblitz/raspiblitz#2378]
With sha256 the menu width I needed to change from 74 to 100.
I will need to be editing the bonus.services files anyway to my Tor PR, should I change this too?

![temp2](https://user-images.githubusercontent.com/69700936/123509158-6b958700-d663-11eb-87e4-c685d7a27a75.png)
![temp1](https://user-images.githubusercontent.com/69700936/123509228-bc0ce480-d663-11eb-8139-258aa7cd1168.png)

**Comment by openoms** [comment]
The question is what does the browser show so we can double check?

**Comment by nyxnor** [comment]
The browser shows both

**Comment by openoms** [comment]
Good, then the shorter is easier to verify.

**Comment by nyxnor** [comment]
Yes, but it is broken https://crypto.stackexchange.com/questions/3690/why-is-sha-1-considered-broken
About easier to verify, when I view the certificate, I Ctrl+F and paste the fingerprint to see if it matches, better than checking by eye.

**Comment by nyxnor** [comment]
![temp](https://user-images.githubusercontent.com/69700936/123525315-36b71d80-d6bf-11eb-8ff3-c06e8fd85b51.png)
from torproject.org with Tor Browser, it is right above.

from stackexchange with FIrefox fork
![temp](https://user-images.githubusercontent.com/69700936/123525392-d2488e00-d6bf-11eb-8c10-f15ffc729ce7.png)

**Comment by openoms** [comment]
How about braking the hash in two lines? Would that make it difficult to copy?

Also check what happens in the default size terminal windows. If 100 character width fits in there there is no problem. If it is wrapped then needs more lines (height) or might be cut off together with the text underneath.

**Comment by nyxnor** [comment]
with 74 (default) It brakes to the second line, but now I cant copy and paste the certificate, as there is blank space between the first and second line.
100 width is the right width for sha256.

![temp](https://user-images.githubusercontent.com/69700936/123526211-d24b8c80-d6c5-11eb-98c3-91bce8619ce5.png)


The space between the lines when searching
![temp](https://user-images.githubusercontent.com/69700936/123526367-eb087200-d6c6-11eb-9885-d4afd39ff48c.png)

**Comment by openoms** [comment]
The problem is when the terminal window is not fullscreen, but small default.

part of the text will be cut off then.

We usually limit the text (even in git commits) to 80 characters for readability in these situations.

The 74 character + window borders just fit in.

I think dialog is already a 'just about acceptable UX' so shouldn't really grow all info windows like that to 100.

Copying only part of the hash should do it for the ones following this practice, what do you think?

**Comment by nyxnor** [comment]
Yes, copying part of that hash is a workaround, not perfect per se.

What could be done is echoing the sha256 fingerprint in the terminal, just like the tor address is when calling `bonus.service.sh menu`
I was not aiming to include the fingerprint in the  `tor.onion-service.sh credentials $service` script cause it is not needed for Tor, but it can be used for that, let me show you how it can help connecting to Tor Browser mobile

**Comment by nyxnor** [comment]
![temp](https://user-images.githubusercontent.com/69700936/123526723-4fc4cc00-d6c9-11eb-88ad-55ecccbc5f47.png)

Now you can just scan the QR code to connect to the service, this will show my apache2 debian default page.
The fingerprint was hardcoded, as I dont use tls for that domain, but the function is ready in the script.

Unfortunately, Tor Browser mobile does not work with Client Auth yet, so I need to warn about this https://gitlab.torproject.org/tpo/applications/tor-browser/-/issues/31672. But you can still scan the QR code on mobile and open in Tor Browser android as normal.

**Comment by nyxnor** [comment]
@rootzoll Even though you marked for v1.8, reading this I prefer to close, open at will.
Reason:
* sha256 is larger than 80 for the msgbox, openoms recommended using just up to 80.
* web ui solves this, the fingerprint can be presented there.

**Comment by openoms** [comment]
> ![temp](https://user-images.githubusercontent.com/69700936/123526723-4fc4cc00-d6c9-11eb-88ad-55ecccbc5f47.png)
> 
> Now you can just scan the QR code to connect to the service, this will show my apache2 debian default page.
> The fingerprint was hardcoded, as I dont use tls for that domain, but the function is ready in the script.
> 
> Unfortunately, Tor Browser mobile does not work with Client Auth yet, so I need to warn about this https://gitlab.torproject.org/tpo/applications/tor-browser/-/issues/31672. But you can still scan the QR code on mobile and open in Tor Browser android as normal.

Don't get me wrong, I do prefer the info shown on this screenshot over the 80 characters insecure fingerprint in whiptail. Re-opening to be implemented non-urgently.
